streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java

```diff
@@ -114,6 +125,17 @@
         this.parentDir = parentDir;
     }
 
+    RocksDBStore(final String name,
+                 final RocksDBMetricsRecorder metricsRecorder) {
+        this(name, DB_FILE_DIR, metricsRecorder);
+    }
+
```

Comment:
The `RocksDBMetricsRecorder` instance passed to the `RocksDBStore` will record the metrics. The recorder is not created inside the constructor, because segmented state stores need to use the same recorder for all of their segments. Each segment of the same segmented state store pass in the same recorder.

Reply:
I am a little late, but following this argument the constructor above should not create an instance of `RocksDBMetricsRecorder` either, but it should rather be an parameter of the constructor?

Reply:
The constructor above is for RocksDB instances that are not segments, thus each of them has its own metrics recorder. The constructor above is for RocksDB instances that are segments, thus the metric recorder is injected. My comment above refers to the below constructor.